test/x86/sse4.2.c

```diff
@@ -24,9 +24,116 @@
 #define SIMDE_TESTS_CURRENT_ISAX sse4_2
 #include <test/x86/test-x86-internal.h>
 #include <simde/x86/sse4.2.h>
+#include <assert.h>
 
 #if defined(SIMDE_X86_SSE4_2_NATIVE) || defined(SIMDE_NO_NATIVE) || defined(SIMDE_ALWAYS_BUILD_NATIVE_TESTS)
 
+static MunitResult
+test_simde_mm_odd_cmpestra(const MunitParameter params[], void* data) {
+  (void) params;
+  (void) data;
+
+  const struct {
+    simde__m128i a;
+    int la;
+    simde__m128i b;
+    int lb;
+    const int imm8;
+    int r;
+  } test_vec[8] = {
+
+  };
+
+  printf("\n");
+  for (size_t i = 0 ; i < (sizeof(test_vec) / (sizeof(test_vec[0]))) ; i++) {
+    simde__m128i_private a, b;
+    int la, lb, r;
+    const int imm8 = (munit_rand_int_range(0, UINT8_MAX) | 1);
+
+    munit_rand_memory(sizeof(a), (uint8_t*) &a);
+    munit_rand_memory(sizeof(b), (uint8_t*) &b);
+    la = munit_rand_int_range(0, 128/16);
+    lb = munit_rand_int_range(0, 128/16);
+
+    r = simde_mm_cmpestra(simde__m128i_from_private(a), la, simde__m128i_from_private(b), lb, imm8);
+
+    printf("    { simde_mm_set_epi16(INT16_C(%6" PRId16 "), INT16_C(%6" PRId16 "), INT16_C(%6" PRId16 "), INT16_C(%6" PRId16 "),\n"
+	   "                         INT16_C(%6" PRId16 "), INT16_C(%6" PRId16 "), INT16_C(%6" PRId16 "), INT16_C(%6" PRId16 ")),\n",
+           a.i16[7], a.i16[6], a.i16[5], a.i16[4], a.i16[3], a.i16[2], a.i16[1], a.i16[0]);
+    printf("      %d ,\n",la);
+    printf("      simde_mm_set_epi16(INT16_C(%6" PRId16 "), INT16_C(%6" PRId16 "), INT16_C(%6" PRId16 "), INT16_C(%6" PRId16 "),\n"
+	   "                         INT16_C(%6" PRId16 "), INT16_C(%6" PRId16 "), INT16_C(%6" PRId16 "), INT16_C(%6" PRId16 ")),\n",
+           b.i16[7], b.i16[6], b.i16[5], b.i16[4], b.i16[3], b.i16[2], b.i16[1], b.i16[0]);
+    printf("      %d ,\n",lb);
+    printf("      %d ,\n",imm8);
```

Comment:
> I think the easiest way would probably be to add another level generator. basically, write a generator that generates the code that you have here, but the end result should be something like

As we have only 8 cases in test_vec , I should be able to do this by manually setting value of imm8 and after take different cases from different instances of generation and put the in test_vec. Is that correct?

Reply:
I haven't changed it as I don't understand it properly yet.

Reply:
Yes, you could do it manually, but that could get tedious pretty fast when you think about the other functions, and you should be able to mostly copy & paste the test case generation code for the other functions.

Based on the _SIDD_* macros, you have character size (2), signed/unsigned (2), comparisons (4), polarity (2), least/most significant (2), and bit/unit mask (2).  I haven't looked at how many apply to each function, but conceivably you could have 2*2*4*2*2*2 = 80 cases to cover.

If you want to do it manually that's okay with me.  Maybe do the first few manually while you get a feel for what is required, then if you think it would make sense later to script it you can.  If you never get to the point where you think scripting would make sense then manually it is :)

Reply:
fair enough, I will write generator code.
Thank You